File: backend/api/internal/domain/domain.go
package domain

type ID int64

type PrivacySettings struct {
	AllowNearbyVisible  bool `json:"allowNearbyVisible"`
	AllowStrangerInvite bool `json:"allowStrangerInvite"`
	AllowComment        bool `json:"allowComment"`
	ShowOwnerName       bool `json:"showOwnerName"`
	ShowCity            bool `json:"showCity"`
	NotificationEnabled bool `json:"notificationEnabled"`
}

type User struct {
	ID        ID              `json:"id"`
	OpenID    string          `json:"openid,omitempty"`
	Nickname  string          `json:"nickname"`
	AvatarURL string          `json:"avatarUrl"`
	City      string          `json:"city"`
	Status    string          `json:"status"`
	Privacy   PrivacySettings `json:"privacy"`
}

type Pet struct {
	ID              ID       `json:"id"`
	UserID          ID       `json:"userId"`
	Name            string   `json:"name"`
	AvatarURL       string   `json:"avatarUrl"`
	Type            string   `json:"type"`
	Breed           string   `json:"breed"`
	Gender          string   `json:"gender"`
	Birthday        string   `json:"birthday,omitempty"`
	Weight          float64  `json:"weight,omitempty"`
	Sterilized      bool     `json:"sterilized"`
	VaccineStatus   string   `json:"vaccineStatus"`
	PersonalityTags []string `json:"personalityTags"`
	InterestTags    []string `json:"interestTags"`
	Description     string   `json:"description"`
	IsDefault       bool     `json:"isDefault"`
	Visible         bool     `json:"visible"`
	Status          string   `json:"status"`
	CreatedAt       string   `json:"createdAt"`
	UpdatedAt       string   `json:"updatedAt"`
	OwnerName       string   `json:"ownerName,omitempty"`
	OwnerAvatarURL  string   `json:"ownerAvatarUrl,omitempty"`
	DistanceText    string   `json:"distanceText,omitempty"`
	ActiveText      string   `json:"activeText,omitempty"`
}

type NearbyPet struct {
	Pet
	DistanceValue       int  `json:"distanceValue"`
	CanInvite           bool `json:"canInvite"`
	CommonInterestCount int  `json:"commonInterestCount"`
}

type Invite struct {
	ID           ID     `json:"id"`
	FromUserID   ID     `json:"fromUserId"`
	FromPetID    ID     `json:"fromPetId"`
	ToUserID     ID     `json:"toUserId"`
	ToPetID      ID     `json:"toPetId"`
	Type         string `json:"type"`
	Title        string `json:"title"`
	Description  string `json:"description"`
	LocationName string `json:"locationName"`
	MeetTime     string `json:"meetTime"`
	Status       string `json:"status"`
	CreatedAt    string `json:"createdAt"`
	UpdatedAt    string `json:"updatedAt"`
	FromPet      *Pet   `json:"fromPet,omitempty"`
	ToPet        *Pet   `json:"toPet,omitempty"`
}

type Post struct {
	ID           ID       `json:"id"`
	UserID       ID       `json:"userId"`
	PetID        ID       `json:"petId"`
	Pet          *Pet     `json:"pet,omitempty"`
	Content      string   `json:"content"`
	Images       []string `json:"images"`
	LocationName string   `json:"locationName"`
	City         string   `json:"city"`
	TopicTags    []string `json:"topicTags"`
	Visibility   string   `json:"visibility"`
	LikeCount    int      `json:"likeCount"`
	CommentCount int      `json:"commentCount"`
	Liked        bool     `json:"liked"`
	Status       string   `json:"status"`
	CreatedAt    string   `json:"createdAt"`
	UpdatedAt    string   `json:"updatedAt"`
}

type Comment struct {
	ID        ID     `json:"id"`
	PostID    ID     `json:"postId"`
	UserID    ID     `json:"userId"`
	PetID     ID     `json:"petId,omitempty"`
	Pet       *Pet   `json:"pet,omitempty"`
	Content   string `json:"content"`
	Status    string `json:"status"`
	CreatedAt string `json:"createdAt"`
	UpdatedAt string `json:"updatedAt"`
}

type Report struct {
	ID             ID       `json:"id"`
	ReporterUserID ID       `json:"reporterUserId"`
	TargetType     string   `json:"targetType"`
	TargetID       ID       `json:"targetId"`
	Reason         string   `json:"reason"`
	Description    string   `json:"description"`
	Images         []string `json:"images"`
	Status         string   `json:"status"`
	CreatedAt      string   `json:"createdAt"`
}

type Block struct {
	ID            ID     `json:"id"`
	UserID        ID     `json:"userId"`
	BlockedUserID ID     `json:"blockedUserId"`
	Reason        string `json:"reason"`
	CreatedAt     string `json:"createdAt"`
}

type PageResult[T any] struct {
	List     []T `json:"list"`
	Page     int `json:"page"`
	PageSize int `json:"pageSize"`
	Total    int `json:"total"`
}

File: frontend/pet-social-mini/src/types/domain.ts
export type ID = number

export type UserStatus = 'normal' | 'disabled' | 'banned' | 'deleted'
export type PetType = 'dog' | 'cat' | 'rabbit' | 'bird' | 'other'
export type Gender = 'unknown' | 'male' | 'female'
export type VaccineStatus = 'unknown' | 'partial' | 'completed'
export type InviteType = 'walk' | 'play' | 'park' | 'coffee' | 'photo' | 'event' | 'custom'
export type InviteStatus =
  | 'pending'
  | 'accepted'
  | 'rejected'
  | 'cancelled'
  | 'expired'
  | 'completed'
  | 'reported'
export type FeedType = 'recommend' | 'nearby' | 'following' | 'mine'
export type PostVisibility = 'public' | 'nearby' | 'followers' | 'private'
export type ReportTargetType = 'user' | 'pet' | 'post' | 'comment' | 'invite'

export interface PrivacySettings {
  allowNearbyVisible: boolean
  allowStrangerInvite: boolean
  allowComment: boolean
  showOwnerName: boolean
  showCity: boolean
  notificationEnabled: boolean
}

export interface User {
  id: ID
  openid?: string
  nickname: string
  avatarUrl: string
  city: string
  status: UserStatus
  phone?: string
  gender?: Gender
  lastLoginAt?: string
  privacy: PrivacySettings
}

export interface Pet {
  id: ID
  userId: ID
  name: string
  avatarUrl: string
  type: PetType
  breed: string
  gender: Gender
  birthday?: string
  weight?: number
  sterilized: boolean
  vaccineStatus: VaccineStatus
  personalityTags: string[]
  interestTags: string[]
  description: string
  isDefault: boolean
  visible: boolean
  status: 'normal' | 'hidden' | 'deleted'
  createdAt: string
  updatedAt: string
  ownerName?: string
  ownerAvatarUrl?: string
  distanceText?: string
  activeText?: string
  isFollowed?: boolean
}

export interface NearbyFilter {
  type: 'all' | PetType
  distance: 1000 | 3000 | 5000 | 10000
  gender: 'all' | 'male' | 'female'
  age: 'all' | 'lt1' | '1to3' | 'gt3'
  personalityTags: string[]
  interestTags: string[]
  canInviteOnly: boolean
  activeOnly: boolean
}

export interface NearbyPet extends Pet {
  distanceValue: number
  distanceText: string
  activeText: string
  canInvite: boolean
  commonInterestCount: number
}

export interface Invite {
  id: ID
  fromUserId: ID
  fromPetId: ID
  toUserId: ID
  toPetId: ID
  type: InviteType
  title: string
  description: string
  locationName: string
  meetTime: string
  status: InviteStatus
  createdAt: string
  updatedAt: string
  fromPet?: Pet
  toPet?: Pet
}

export interface Post {
  id: ID
  userId: ID
  petId: ID
  pet?: Pet
  content: string
  images: string[]
  locationName: string
  city: string
  topicTags: string[]
  visibility: PostVisibility
  likeCount: number
  commentCount: number
  liked: boolean
  status: 'normal' | 'hidden' | 'deleted'
  createdAt: string
  updatedAt: string
}

export interface Comment {
  id: ID
  postId: ID
  userId: ID
  petId?: ID
  pet?: Pet
  content: string
  status: 'normal' | 'hidden' | 'deleted'
  createdAt: string
  updatedAt: string
}

export interface Report {
  id: ID
  reporterUserId: ID
  targetType: ReportTargetType
  targetId: ID
  reason: string
  description: string
  images: string[]
  status: 'pending' | 'accepted' | 'rejected'
  createdAt: string
}

export interface Block {
  id: ID
  userId: ID
  blockedUserId: ID
  reason: string
  createdAt: string
}

export interface PageResult<T> {
  list: T[]
  page: number
  pageSize: number
  total: number
}

export interface LoginResult {
  token: string
  user: User
  hasPet: boolean
}

export interface LocationPayload {
  latitude: number
  longitude: number
  city: string
  district: string
}

export interface CreatePetPayload {
  name: string
  avatarUrl: string
  type: PetType
  breed: string
  gender: Gender
  birthday?: string
  weight?: number
  sterilized: boolean
  vaccineStatus: VaccineStatus
  personalityTags: string[]
  interestTags: string[]
  description: string
  visible?: boolean
}

export interface CreateInvitePayload {
  fromPetId: ID
  toPetId: ID
  type: InviteType
  title: string
  description: string
  locationName: string
  meetTime: string
}

export interface CreatePostPayload {
  petId: ID
  content: string
  images: string[]
  locationName: string
  topicTags: string[]
  visibility: PostVisibility
}

export interface CreateReportPayload {
  targetType: ReportTargetType
  targetId: ID
  reason: string
  description: string
  images: string[]
}
